packages/end-to-end/cypress/support/commands.js

```diff
@@ -1,25 +1,159 @@
-// create various custom commands and overwrite
-// existing commands.
-//
-// For more comprehensive examples of custom
-// commands please read more here:
-// https://on.cypress.io/custom-commands
-// ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
-//
-//
-// -- This is a child command --
-// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
-//
-//
-// -- This is a dual command --
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
\ No newline at end of file
+import 'cypress-react-selector';
+
+Cypress.Commands.add('waitUntilAnyVisible', (selector1, selector2, timeout = 10000) => {
+  const startTime = new Date().getTime();
+
+  const checkVisibility = () => {
+    return cy.get('body', { log: false }).then(($body) => {
+      if ($body.find(selector1).is(':visible')) {
+        return cy.wrap($body.find(selector1), { log: false });
+      } else if ($body.find(selector2).is(':visible')) {
+        return cy.wrap($body.find(selector2), { log: false });
+      } else if (new Date().getTime() - startTime < timeout) {
+        // If timeout hasn't passed, recursively call the function
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        return cy.wait(100, { log: false }).then(checkVisibility);
+      } else {
+        // If timeout has passed, throw an error
+        throw new Error(
+          `Neither ${selector1} nor ${selector2} became visible within ${timeout}ms.`,
+        );
+      }
+    });
+  };
+
+  return checkVisibility();
+});
+
+Cypress.Commands.add(
+  'loginOrCreateAccount',
+  (email, password, fullName, language, crop_menu_name) => {
+    //Login page
+    cy.get('[data-cy=email]').type(email);
+    cy.intercept('GET', '**/login/user/' + email, (req) => {
+      delete req.headers['if-none-match'];
+    }).as('emailLogin');
+    cy.get('[data-cy=continue]').should('exist').and('be.enabled').click();
+
+    cy.waitUntilAnyVisible('[data-cy=createUser-email]', '[data-cy=enterPassword-password]');
+
+    // Check if we are on pasword page or create account and act accordingly
+    cy.get('[data-cy=createUser-email]', { timeout: 0 }).then(($element) => {
+      if ($element.is(':visible')) {
+        cy.get('[data-cy=createUser-fullName]').type(fullName);
+
+        // cy.get('[id$=react-select-3-input]').type(language+'{enter}');
+        cy.contains('English')
+          .parent()
+          .find('input')
+          .type(language + '{enter}');
+
+        cy.get('[data-cy=createUser-password]').type(password);
+        cy.get('[data-cy=createUser-create]').should('exist').and('be.enabled').click();
+
+        cy.intercept('POST', '**/user').as('createUser');
```

Comment:
Is the intention behind this intercept to wait for the request to resolve? I think we'd need to add `cy.wait(@createUser)` below this line for it to wait if that's the case.